### PR TITLE
feature: commands to access env resources

### PR DESCRIPTION
**Pull request by Mugenor**
- Add cmd to open env resources: grafana, grafana/continuous-load

### CODE REVIEW

File: pkg/browser/open.go [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+			cmd = "cmd.exe"
+			args = []string{"/c", "start", url}
+		} else {
+			cmd = "xdg-open"
+			args = []string{url}
+		}
+	}
+	if len(args) > 1 {
+		// args[0] is used for 'start' command argument, to prevent issues with URLs starting with a quote
+		args = append(args[:1], append([]string{""}, args[1:]...)...)
```
Comment (SergeiSizov) [inline pkg/browser/open.go:33]: my brain feels pain understanding this line

File: pkg/browser/open.go [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+	}
+	return exec.Command(cmd, args...).Start()
+}
+
+// isWSL checks if the Go program is running inside Windows Subsystem for Linux
+func isWSL() bool {
+	releaseData, err := exec.Command("uname", "-r").Output()
+	if err != nil {
```
Comment (SergeiSizov) [inline pkg/browser/open.go:41]: can we handle the error somehow? e.g. file not found?

File: pkg/cmd/env/open.go [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+		environment.DirFromCPlatformRepoPath(cfg.Repositories.CPlatform.Value),
+		opts.Environment,
+	)
+	if err != nil {
+		return err
+	}
+	if env == nil {
+		return fmt.Errorf("environment %s not found", opts.Environment)
+	}
+	if err := corectlenv.OpenResource(corectlenv.ResourceType(opts.Resource), env); err != nil {
```
Comment (SergeiSizov) [inline pkg/cmd/env/open.go:57]: maybe just 
```
return corectlenv.OpenResource(corectlenv.ResourceType(opts.Resource), env)
```

File: pkg/env/resources.go [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+	}
+}
+
+func openGrafana(env *environment.Environment) error {
+	url := "https://grafana." + env.InternalServices.Domain
+	return browser.OpenURL(url)
+}
+
+func openGrafanaContinuousLoad(env *environment.Environment) error {
+	url := "https://grafana." + env.InternalServices.Domain + "/d/zDpLnqaMz/continuous-load?orgId=1&refresh=5s"
```
Comment (SergeiSizov) [inline pkg/env/resources.go:44]: maybe Sprintf would be nicer?

**Review by SergeiSizov — COMMENTED** [review]
what about unit tests?

File: pkg/browser/open.go (hunk 2 above)
Comment (Mugenor) [reply pkg/browser/open.go:41]: I'm not sure what to handle here. It expects that this command will work in WSL. If doesn't work, it's not WSL

File: pkg/cmd/env/open.go (hunk 3 above)
Comment (Mugenor) [reply pkg/cmd/env/open.go:57]: I have a fixed feelings about this one. It's less code, true, but it looks like this function have to return some result, when in the reality it just returns error.
When I looking at the current version of code, I have less questions

File: pkg/env/resources.go (hunk 4 above)
Comment (Mugenor) [reply pkg/env/resources.go:44]: I like this version better :)
I don't have to match '%' parameters in string with parameters passed to the sprintf

**Comment by Mugenor** [comment]
About unit tests: there is zero logic, I'm not sure what to test here
Maybe URL construction, but the code is too straightforward

File: pkg/browser/open.go (hunk 2 above)
Comment (SergeiSizov) [reply pkg/browser/open.go:41]: yeah, I would also support adding some context or logging.

File: pkg/cmd/env/open.go (hunk 3 above)
Comment (SergeiSizov) [reply pkg/cmd/env/open.go:57]: then maybe wrap the error with context?

File: pkg/cmd/env/open.go [hunk 5]
```diff
@@ -0,0 +1,61 @@
+package env
+
+import (
+	"fmt"
+	"github.com/coreeng/corectl/pkg/cmdutil/config"
+	"github.com/coreeng/corectl/pkg/cmdutil/userio"
+	corectlenv "github.com/coreeng/corectl/pkg/env"
+	"github.com/coreeng/developer-platform/pkg/environment"
+	"github.com/spf13/cobra"
+	"strings"
```
[the remark this answers is not in the document]
Comment (Mugenor) [reply pkg/cmd/env/open.go:10]: What is the problem?
And I don't think I understand what you are suggesting

File: pkg/cmd/env/open.go [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+		return err
+	}
+	env, err := environment.FindByName(
+		environment.DirFromCPlatformRepoPath(cfg.Repositories.CPlatform.Value),
+		opts.Environment,
+	)
+	if err != nil {
+		return err
+	}
+	if env == nil {
```
[the remark this answers is not in the document]
Comment (Mugenor) [reply pkg/cmd/env/open.go:54]: This part of the code lives in developer-platform. 
We can change the logic, but probably not as part of this task

File: pkg/browser/open.go (hunk 2 above)
Comment (Mugenor) [reply pkg/browser/open.go]: I replaced this code with the lib call

File: pkg/browser/open.go (hunk 1 above)
Comment (Mugenor) [reply pkg/browser/open.go]: I replaced this code with the lib call

File: pkg/cmd/env/open.go (hunk 3 above)
Comment (Mugenor) [reply pkg/cmd/env/open.go:57]: Added some context

File: pkg/browser/open.go [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+import (
+	"os/exec"
+	"runtime"
+	"strings"
+)
+
+// OpenURL opens the specified URL in the default browser of the user.
+// Snippet source: https://gist.github.com/sevkin/9798d67b2cb9d07cb05f89f14ba682f8?permalink_comment_id=5084817#gistcomment-5084817
+func OpenURL(url string) error {
+	var cmd string
```
[the remark this answers is not in the document]
Comment (Mugenor) [reply pkg/browser/open.go]: I replaced this code with the lib call

File: pkg/browser/open.go [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+		args = []string{"/c", "start"}
+	case "darwin":
+		cmd = "open"
+		args = []string{url}
+	default: // "linux", "freebsd", "openbsd", "netbsd"
+		if isWSL() {
+			cmd = "cmd.exe"
+			args = []string{"/c", "start", url}
+		} else {
+			cmd = "xdg-open"
```
[the remark this answers is not in the document]
Comment (Mugenor) [reply pkg/browser/open.go]: I replaced this code with the lib call

File: pkg/browser/open.go [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+	if len(args) > 1 {
+		// args[0] is used for 'start' command argument, to prevent issues with URLs starting with a quote
+		args = append(args[:1], append([]string{""}, args[1:]...)...)
+	}
+	return exec.Command(cmd, args...).Start()
+}
+
+// isWSL checks if the Go program is running inside Windows Subsystem for Linux
+func isWSL() bool {
+	releaseData, err := exec.Command("uname", "-r").Output()
```
[the remark this answers is not in the document]
Comment (Mugenor) [reply pkg/browser/open.go]: I replaced this code with the lib call

**Review by SergeiSizov — APPROVED** [review]
lgtm

File: pkg/cmd/env/open.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (Mugenor) [reply pkg/cmd/env/open.go:10]: I use `strings` just to join all possible resource options and output it as `help`

**Pull request merged**